content/guides/spring-integration-lock/index.md

```diff
@@ -0,0 +1,366 @@
+---
+date: '2021-10-22'
+lastmod: '2021-10-22'
+title: Working With the Spring Distributed Lock
+linkTitle: Working With the Spring Distributed Lock
+description: A simple demonstration of how to implement a Spring Distributed Lock with in your Spring application
+patterns:
+  - API
+tags:
+  - Spring Boot
+  - Spring Integration
+  - API
+  - Getting Started
+  - Spring
+  - Microservices
+team:
+  - Eric Standley
+weight: 2
+oldPath: '/content/guides/spring/spring-distributed-lock.md'
+aliases:
+  - '/guides/spring/spring-distributed-lock'
+level1: Building Modern Applications
+level2: Frameworks and Languages
+---
+
+When designing microservices, you want to create code to handle multiple instances of services to run at the same time. When you only want to process something one time, or do something similar to the [Leader Pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/leader-election), you cannot synchronize if the code runs in different pods. As a result, you have to use an external method that is fraught with pitfalls during implementation[^1]. 
```

Comment:
`you want to create code to handle multiple instances of services to run at the same time.` I don't think this is quite what I meant, since this lock should only be used with a single service. I meant to start with the larger idea of when designing any micro service but then talk about the singular service we're going to design in this guide

Reply:
`When creating microservices, the typical pattern is to design services that can handle multiple instances running at the same time. However, there are instances where only a single service is preferable. For example, in the [Leader Pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/leader-election), you cannot...`
Better?

Reply:
@anthonyvetter, I like how the sentence reads. Yes, I think it's better!